software/controller/test/debug/vars_test.cpp

```diff
@@ -150,3 +155,97 @@ TEST(DebugVar, Registration) {
   EXPECT_EQ(&var2, Registry::singleton().find(var2.id()));
   EXPECT_EQ(nullptr, Registry::singleton().find(12345));
 }
+
+TEST(DebugVar, NonVolatile) {
```

Comment:
This is one big test case. Maybe have a few smaller test cases? Especially if you split the headers, maybe this var gets its own test suite? Also, would like to see something like how it would behave if it were unlinked. There is no way to "gracefully" segfault on bare metal.